pkg/detectors/apimatic/apimatic.go

```diff
@@ -23,8 +23,7 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"apimatic"}) + `\b([a-zA-Z0-9]{3,20}@[a-zA-Z0-9]{2,12}.[a-zA-Z0-9]{2,5})\b`)
-	passPat = regexp.MustCompile(detectors.PrefixRegex([]string{"apimatic"}) + `\b([a-z0-9-\S]{8,32})\b`)
+	apiKeyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"apimatic", "apikey"}) + `\b([a-zA-Z0-9_-]{64})\b`)
```

Comment:
I somehow discovered that APIMatic is no longer using these for authentication and now utilizing an `Auth Key`. Additionally, the API we were using earlier appears to no longer exist; it may have been available before, but the path seems to have changed in their documentation. I confirmed these updates through some local testing.

Documentation links are added in code comments, if you wanna take a look 🕵🏻

Reply:
Good Test:
![Screenshot from 2024-10-31 19-25-26](https://github.com/user-attachments/assets/cb9dfec9-ffb6-4556-ad5a-2137b0102fde)
Bad Test:
![Screenshot from 2024-10-31 19-26-53](https://github.com/user-attachments/assets/491e2b34-6e9f-4560-84f4-b17d63266d8d)